File: WeeklyChallenge2022.playground/Pages/Challenge0.xcplaygroundpage/Contents.js
/* 
* EL FAMOSO "FIZZ BUZZ"
 * Fecha publicación enunciado: 27/12/21
 * Fecha publicación resolución: 03/01/22
 * Dificultad: FÁCIL
 *
 * Enunciado: Escribe un programa que muestre por consola (con un print) los números de 1 a 100 (ambos incluidos y con un salto de línea entre cada impresión), sustituyendo los siguientes:
 * - Múltiplos de 3 por la palabra "fizz".
 * - Múltiplos de 5 por la palabra "buzz".
 * - Múltiplos de 3 y de 5 a la vez por la palabra "fizzbuzz".
 *
 * Información adicional:
 * - Usa el canal de nuestro discord (https://mouredev.com/discord) "🔁reto-semanal" para preguntas, dudas o prestar ayuda a la comunidad.
 * - Puedes hacer un Fork del repo y una Pull Request al repo original para que veamos tu solución aportada.
 * - Revisaré el ejercicio en directo desde Twitch el lunes siguiente al de su publicación.
 * - Subiré una posible solución al ejercicio el lunes siguiente al de su publicación.
 *  */

// Solución 
/* Función que devuelve el número o la palabra fizz, buzz o fizzbuzz.
    De tal manera que si el número es múltiplo de 3 devuelva fizz,
    si el numero es múltiplo de 5 devuelva buzz y si es múltiplo de 3 y 5 devuelva fizzbuzz
    Todo ello se lo añadimos a una lista que devolvemos al final de la función */
const fizzBuzz = (number) => {
  let lista_numeros = [];
  for (let index = 1; index < number + 1; index++) {
    if (index % 3 === 0 && index % 5 === 0) {
      lista_numeros.push("FizzBuzz");
    } else if (index % 3 === 0) {
      lista_numeros.push("Fizz");
    } else if (index % 5 === 0) {
      lista_numeros.push("Buzz");
    } else {
      lista_numeros.push(index);
    }
  }
  return lista_numeros;
};

// Realizamos un bucle for para recorrer la lista que nos devuelve la función fizzbuzz
for (const iterator of fizzBuzz(100)) {
  console.log(iterator);
}

File: WeeklyChallenge2022.playground/Pages/Challenge4.xcplaygroundpage/Contents.py
"""
* Reto #4
 * ÁREA DE UN POLÍGONO
 * Fecha publicación enunciado: 24/01/22
 * Fecha publicación resolución: 31/01/22
 * Dificultad: FÁCIL
 *
 * Enunciado: Crea UNA ÚNICA FUNCIÓN (importante que sólo sea una) que sea capaz de calcular y retornar el área de un polígono.
 * - La función recibirá por parámetro sólo UN polígono a la vez.
 * - Los polígonos soportados serán Triángulo, Cuadrado y Rectángulo.
 * - Imprime el cálculo del área de un polígono de cada tipo.
 *
 * Información adicional:
 * - Usa el canal de nuestro discord (https://mouredev.com/discord) "🔁reto-semanal" para preguntas, dudas o prestar ayuda a la comunidad.
 * - Puedes hacer un Fork del repo y una Pull Request al repo original para que veamos tu solución aportada.
 * - Revisaré el ejercicio en directo desde Twitch el lunes siguiente al de su publicación.
 * - Subiré una posible solución al ejercicio el lunes siguiente al de su publicación.
 * 
"""

def areaPoligono(poligono: str, area: int, altura: int):
    try:
        if poligono == "triangulo":
            return area * altura / 2
        elif poligono == "cuadrado":
            return area * area
        elif poligono == "rectangulo":
            return area * altura
        else:
            return "No se ha encontrado el polígono"
    except:
        return "Error"
    
print(areaPoligono("triangulo", 20, 10))
print(areaPoligono("cuadrado", 20, 10))
print(areaPoligono("rectangulo", 20, 10))